bin/generator/src/main.cpp

```diff
@@ -0,0 +1,85 @@
+#include <boost/json.hpp>
+#include <boost/json/src.hpp>
+#include <boost/program_options.hpp>
+#include <fstream>
+#include <iostream>
+#include <string>
+
+#include "generator.h"
+#include "zkevm_framework/data_types/block.hpp"
+
+int main(int argc, char* argv[]) {
+    boost::program_options::options_description options_desc("Block generator");
+
+    // clang-format off
+    options_desc.add_options()("help,h", "Display help message")
+            ("version,v", "Display version")
+            ("input-file,i", boost::program_options::value<std::string>(), "Input block in json format")
+            ("output-file,o", boost::program_options::value<std::string>(), "Generated block");
+            ;
+    // clang-format on
+
+    boost::program_options::variables_map vm;
+    try {
+        boost::program_options::store(
+            boost::program_options::command_line_parser(argc, argv).options(options_desc).run(),
+            vm);
+        boost::program_options::notify(vm);
+    } catch (const boost::program_options::unknown_option& e) {
+        std::cerr << "Invalid command line argument: " << e.what() << std::endl;
+        std::cout << options_desc << std::endl;
+        return 1;
+    }
+
+    if (vm.count("help")) {
+        std::cout << options_desc << std::endl;
+        return 0;
+    }
+
+    if (vm.count("version")) {
+#ifdef GENERATOR_VERSION
+#define xstr(s) str(s)
+#define str(s) #s
+        std::cout << xstr(GENERATOR_VERSION) << std::endl;
+#else
+        std::cout << "Version is not defined" << std::endl;
+#endif
+        return 0;
+    }
+
+    if (!vm.count("input-file")) {
+        std::cerr << "Input file was not provided" << std::endl;
+        return 1;
+    }
+    const auto& input_file_name = vm["input-file"].as<std::string>();
+    std::string output_file_name = "block.out";
+    if (vm.count("output-file")) {
+        output_file_name = vm["output-file"].as<std::string>();
+    }
+
+    std::ifstream input_file(input_file_name.c_str());
+    if (!input_file.is_open()) {
+        std::cerr << "Could not open the input file " << input_file_name << std::endl;
+        return 1;
+    }
+    std::optional<boost::json::value> parsed_json =
+        data_types::block_generator::parse_json(input_file);
+    if (!parsed_json) {
+        std::cerr << "Parsing of input file failed: not a JSON format" << std::endl;
+        return 1;
+    }
+
+    std::expected<data_types::Block, std::string> block =
+        data_types::block_generator::generate_block(parsed_json.value());
+    if (!block) {
+        std::cerr << "Error while parsing input file: " << block.error() << std::endl;
+        return 1;
+    }
+
+    std::ofstream output_file(output_file_name);
+    auto serialisation_result = block->serialize(output_file);
+    if (!serialisation_result) {
+        std::cerr << "Serialization error :" << serialisation_result.error().msg;
```

Comment:
Shouldn't we `return 1;` here?

Reply:
Also you can [stream `SerializationError`](https://github.com/NilFoundation/zkEVM-framework/blob/master/libs/data_types/include/zkevm_framework/data_types/errors.hpp#L76) directly into `std::cerr` and it produces slightly better error message that manually extracted `msg`.

Reply:
Added return, but the stream operator was used initially.

Reply:
I think, I don't understand you. I meant that you can write this:

```cpp
std::cerr << serialisation_result.error();
```

Reply:
Sorry, my bad, fixed